aiohttp_devtools/runserver/watch.py

```diff
@@ -119,10 +121,33 @@ def _start_dev_server(self) -> None:
         self._process = Process(target=serve_main_app, args=(self._config, tty_path))
         self._process.start()
 
-    def _stop_dev_server(self) -> None:
+    async def _stop_dev_server(self) -> None:
         if self._process.is_alive():
             logger.debug('stopping server process...')
+            if self._config.shutdown_endpoint:  # a workaround for singals not working on Windows
+                url = 'http://localhost:{}{}/shutdown'.format(self._config.main_port, self._config.path_prefix)
```

Comment:
Hmm, how is it being run then? My impression is that it should be using `.cleanup_ctx()`.

Reply:
Unfortunately I don't know, since I am having a bit of trouble wrapping my head around when all the different startup and shutdown handlers are supposed to be called. But perhaps this could be a similar issue as in #566, where apparently `WatchTask.cleanup_ctx` isn't being called either?

Reply:
Strange, I'm no longer getting the problem where the `_app` attribute isn't set, instead the route table doesn't include the shutdown endpoint - I tried looking at `list(self._app.router.routes())` and it only includes `/livereload.js` and `/livereload`.

Reply:
Since everything else is working, I'll mark this resolved for now...

Reply:
Something seems suspicious about this, I'll try and take a look briefly before merging.

Reply:
OK, this is the aux app, not the user's app. That's why we can't get the URL. The user's app is within another function and appears to be unavailable at this point in the code, so we'll stick with this.